### PR TITLE
Fixes #5195: hide highlights when there are none

### LINKED ISSUE #5195: Hide Highlights when there are none.
It would be nice if the new Highlights section did not appear for competitions that have no highlights. Example: https://www.worldcubeassociation.org/competitions/HaNoiBigCube2012#general-info

### PR DESCRIPTION
**Pull request by danieljames-dj**

### CODE REVIEW

File: WcaOnRails/app/views/competitions/_competition_info.html.erb [hunk 1]
```diff
@@ -171,22 +171,23 @@
...
-            <% if main_event.present? %>
-              <p><%=md winners(competition, main_event) %></p>
-            <% end %>
-
-            <p><%=md records(competition) %></p>
-          </div>
-        </dd>
+        <% records = md records(competition) %>
+        <% main_event_message = main_event.present? ? (md winners(competition, main_event)) : "" %>
+        <% if !main_event_message.blank? || !records.blank? %>
```
Comment (viroulep) [inline WcaOnRails/app/views/competitions/_competition_info.html.erb:176]: Here you can keep the markdown rendering within the block by doing something like:
```suggestion
        <% records = records(competition) %>
        <% if main_event.present? || records.present? %>
```
and below
```erb
              <p><%= md(winners(competition, main_event)) if main_event.present? %></p>
              <p><%= md(records) %></p>
```

File: WcaOnRails/app/views/competitions/_competition_info.html.erb [hunk 2]
```diff
@@ -171,22 +171,23 @@
...
-          <div class="collapse" id="highlights_text">
-            <% if main_event.present? %>
-              <p><%=md winners(competition, main_event) %></p>
-            <% end %>
-
-            <p><%=md records(competition) %></p>
-          </div>
-        </dd>
+        <% records = md records(competition) %>
+        <% main_event_message = main_event.present? ? (md winners(competition, main_event)) : "" %>
```
Comment (viroulep) [inline WcaOnRails/app/views/competitions/_competition_info.html.erb:175]: Here `md` is a regular function call (it's just that confusingly enough, in ruby, you can omit parenthesis when the context allow for appropriate parsing...).
So the ternary condition could be written as:
`main_event.present? ? md(winners(competition, main_event)) : ""`
(the opening parenthesis has moved)

**Comment by danieljames-dj** [comment]
Thanks, @viroulep the nitpicks that you suggested definitely makes sense and I would love to apply those changes. I made the changes and force pushed :-)

**Comment by viroulep** [comment]
Thanks a lot!

**Pull request merged**